### PR TITLE
[NETBEANS-774] Disable var hint for anonymous type

### PR DESCRIPTION
**Pull request by rtaneja1**
This was open issue from #509.

### CODE REVIEW

**Comment by rtaneja1** [comment]
@jlahoda, @sdedic, Can you please review.

File: java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java [hunk 1]
```diff
@@ -152,16 +155,25 @@ private static boolean hasDiagnosticErrors(CompilationInfo info, Tree tree) {
...
             if (nct.getClassBody() != null) {
                 return false;
             }
+        } else if (initTreePath.getLeaf().getKind() == Tree.Kind.METHOD_INVOCATION) {
+            //filter anonymous class type return type
+            if (variableTypeMirror.getKind() == TypeKind.DECLARED) {
+                DeclaredType dt = (DeclaredType) variableTypeMirror;
+                TypeElement typeElem = (TypeElement) dt.asElement();
+                ClassSymbol sym = (ClassSymbol) typeElem;
+                if (sym.isAnonymous()) {
```
Comment (jlahoda) [inline java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java]: Please do not use internal javac classes outside of the infrastructure (ideally, outside of java.source* and lib.nbjavac, although I know it is used in spi.java.hints as well).

In this case typeElem.getNestingKind() == NestingKind.ANONYMOUS should work, right?

File: java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java [hunk 2]
```diff
@@ -152,16 +155,25 @@ private static boolean hasDiagnosticErrors(CompilationInfo info, Tree tree) {
...
         TreePath treePath = ctx.getPath();
         TreePath initTreePath = ctx.getVariables().get("$init");  //NOI18N
+        TypeMirror variableTypeMirror = ctx.getInfo().getTrees().getElement(treePath).asType();
 
         if (initTreePath.getLeaf().getKind() == Tree.Kind.NEW_CLASS) {
             NewClassTree nct = ((NewClassTree) initTreePath.getLeaf());
             if (nct.getClassBody() != null) {
                 return false;
             }
+        } else if (initTreePath.getLeaf().getKind() == Tree.Kind.METHOD_INVOCATION) {
```
Comment (jlahoda) [inline java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java]: Not sure if this check is correct:
-why look at the structure of the initializer when we have the variable type? If the variable type contains anonymous, that is a problem, if it does not, all is fine (i.e. why not simple look at variableTypeMirror?). What if the code is like:
var v1 = new Object() {};
var v2 = v1; //v2 is anonymous as well, right?
-what about cases like:
var v = List.of(new Object() {});
?

Comment (rtaneja1) [reply java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java]: Fixed these usecases. Need to handle usecase var v = {new Object()} in the new error hint for array initializer. I'm trying to push that change in same PR.

Comment (rtaneja1) [reply java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java]: Fixed

File: java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java (hunk 1 above)
Comment (rtaneja1) [reply java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java]: Corrected.

**Comment by jlahoda** [comment]
Ok.

**Pull request merged**